### PR TITLE
fix: show selections in ComparisonView on any page (ET-189)

### PR DESCRIPTION
**Pull request by johnkim-det**
## Ticket

ET-189

## Description

Even if selected experiments/runs are not loaded on the current page, the Compare panel should still fetch data for them.

## Test Plan

On Experiments List page (with "New Experiment List" feature flag on):
- Select an experiment
- Open the "Compare" view
- Change the page
- Should display "Metrics", "Hyperparameters", and "Details" tabs with appropriate content
  - Should NOT display "No items selected." message

On Runs List page (with "Flat Runs View" feature flag on):
- Select a run
- Open the "Compare" view
- Change the page
- Should display "Metrics", "Hyperparameters", and "Details" tabs with appropriate content
  - Should NOT display "No items selected." message

## Checklist

- [ ] Changes have been manually QA'd
- [ ] New features have been approved by the corresponding PM
- [ ] User-facing API changes have the "User-facing API Change" label
- [ ] Release notes have been added as a separate file under `docs/release-notes/`
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses have been included for new code which was copied and/or modified from any external code

[DET-123]: https://hpe-aiatscale.atlassian.net/browse/DET-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by ashtonG — COMMENTED** [review]
given that the rest of the codebase supports both types of selections, can we do the work to support that here?

**Comment by ashtonG** [comment]
* i'm seeing an issue where the colors in the comparison view doesn't match the colors in the table. This can be seen by selecting experiments out of table order.
* Furthermore, we should probably include messaging when the user has selected more than 50 experiments/runs (in order to handle the select all branch, use the pagination information to determine if more than 50 experiments/runs were in the selection)
* this wasn't in the ticket, but i'm seeing 431 errors when comparing more than ~20 experiments. this is specifically the searchExperiments endpoint. There's a ticket to switch it to use the post method like we did for searchTrials (ET-602) -- i think we need to block work on this until the other ticket is finished.

File: webui/react/src/components/ComparisonView.tsx [hunk 1]
```diff
@@ -38,13 +49,84 @@ const ComparisonView: React.FC<Props> = ({
...
+  );
+
+  const loadableSelectedRuns = useAsync(async () => {
+    if (runSelection) {
+      const filterFormSet = INIT_FORMSET;
+      try {
+        const filter = getRunIdsFilter(filterFormSet, runSelection);
+        const response = await searchRuns({
+          filter: JSON.stringify(filter),
+          limit: 50,
```
Comment (ashtonG) [inline webui/react/src/components/ComparisonView.tsx:117]: nit: remember to use the selection limit here as well.

**Review by ashtonG — APPROVED** [review]
lgtm

File: webui/react/src/components/ComparisonView.tsx [hunk 2]
```diff
@@ -38,13 +49,84 @@ const ComparisonView: React.FC<Props> = ({
...
+        (runSelection.type === 'ALL_EXCEPT' &&
+          total &&
+          total - runSelection.exclusions.length > SELECTION_LIMIT)
+      ) {
+        return true;
+      }
+      return false;
+    }
+    return false;
+  };
```
[the remark this answers is not in the document]
Comment (johnkim-det) [reply webui/react/src/components/ComparisonView.tsx:85]: ah, that makes sense, will change this to a state boolean that's set based on `pagination.total` in `searchRuns`/`searchExperiments` response.

**Pull request merged**